packages/editor/src/components/rich-text/format-toolbar/link-container.js

```diff
@@ -53,6 +53,22 @@ function isShowingInput( props, state ) {
 	return props.addingLink || state.editLink;
 }
 
+/**
+ * Safely decodes a URI with `decodeURI`. Returns the URI unmodified if
+ * `decodeURI` throws an error.
+ *
+ * @param {string} uri URI to encode.
+ *
+ * @return {string} Encoded URI if possible.
+ */
+function safeDecodeURI( uri ) {
```

Comment:
Should we make it part of `@wordpress/url` package? I see another place where `decodeURI` is used in our code.

`{ decodeURI( samplePermalink ) }`

Reply:
Yes, that's what I'm doing right now.

Reply:
See 
https://github.com/WordPress/gutenberg/pull/10505#issuecomment-428903463